omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,199 @@
+import org.openmrs.module.webservices.rest.web.annotation.RepHandler;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.util.OpenmrsClassLoader;
+
+import java.util.ArrayList;
+
+@Resource(name = RestConstants.VERSION_1 + "/tag", supportedClass = Tag.class, supportedOpenmrsVersions = { "1.11.*",
+        "1.12.*", "2.0.*", "2.1.*" })
+public class TagResource extends DelegatingCrudResource<Tag> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#newDelegate()
+	 */
+	@Override
+	public Tag newDelegate() {
+		return new Tag();
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getByUniqueId(String)
+	 */
+	@Override
+	public Tag getByUniqueId(String uuid) {
+		return getService().getTagByUuid(uuid);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(Object)
+	 */
+	@Override
+	public Tag save(Tag tag) {
+		return getService().saveTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#delete(Object,String, RequestContext)
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		throw new ResourceDoesNotSupportOperationException("delete not allowed on tag");
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		getService().purgeTag(tag);
+	}
+	
+	/**
+	 * Returns the DefaultRepresentation
+	 */
+	@RepHandler(DefaultRepresentation.class)
+	public SimpleObject asDef(Tag delegate) throws ConversionException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("tag");
+		description.addLink(delegate.getObjectType().substring(delegate.getObjectType().lastIndexOf(".") + 1).toLowerCase(),
```

Comment:
You shouldn't be manually building the URI, rest has a way to generate the URI for an object, fetch the resource from the RestService and then call `getUri(Object)`.